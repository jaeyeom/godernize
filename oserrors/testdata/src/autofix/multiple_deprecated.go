package autofix

import (
	"fmt"
	"os"
)

var _ = multipleDeprecated

func multipleDeprecated() {
	var err error
	if os.IsNotExist(err) { // want `os.IsNotExist is deprecated, use errors.Is\(err, fs.ErrNotExist\) instead`
		fmt.Println("File does not exist")
	}
	if os.IsExist(err) { // want `os.IsExist is deprecated, use errors.Is\(err, fs.ErrExist\) instead`
		fmt.Println("File exists")
	}
	if os.IsPermission(err) { // want `os.IsPermission is deprecated, use errors.Is\(err, fs.ErrPermission\) instead`
		fmt.Println("Permission denied")
	}
}
