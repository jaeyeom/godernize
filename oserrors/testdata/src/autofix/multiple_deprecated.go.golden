package autofix

import (
	"fmt"
	"os"
)

var _ = multipleDeprecated

func multipleDeprecated() {
	var err error
	if errors.Is(err, fs.ErrNotExist) { // want `os.IsNotExist is deprecated, use errors.Is\(err, fs.ErrNotExist\) instead`
		fmt.Println("File does not exist")
	}
	if errors.Is(err, fs.ErrExist) { // want `os.IsExist is deprecated, use errors.Is\(err, fs.ErrExist\) instead`
		fmt.Println("File exists")
	}
	if errors.Is(err, fs.ErrPermission) { // want `os.IsPermission is deprecated, use errors.Is\(err, fs.ErrPermission\) instead`
		fmt.Println("Permission denied")
	}
}